### ISSUE TITLE
NSE problem with x?

### ISSUE THREAD

**Issue by mdsumner** [hypertidy/tidync#13]
This in https://github.com/hypertidy/tidync/commit/c2f5703a083f71c01672abb6f13b9691ea4a752f

```
  # Error in hyper_filter(., x = x < 30) : object 'x' not found
  (f <- "/rdsi/PRIVATE/raad/data_local/amps/2013021900_WRF_d2_PCP_f009.nc")
  PCP <- tidync(f) %>% activate(PCP)
  PCP %>% hyper_filter()
 # expect_error(PCP %>% hyper_filter(x = x < 30), "object 'x' not found")
  expect_error(PCP %>% hyper_filter(x = x > 1), "\\(list\\) object cannot be coerced to type 'double'")

```

It's somehow clashing object x with named element x, since "y" works fine.

**Comment by mdsumner** [comment]
I saw another example, with an overwrought pipe line, something like

```R
 estimatebigtime <- x$grid %>% 
    dplyr::filter(grid == active(x)) %>% 
    dplyr::inner_join(x$variable, c("variable" = "name")) %>% 
    dplyr::distinct(dimids) %>% 
    dplyr::inner_join(x$dimension, c("dimids" = "id")) %>% pull(length) %>% prod()

```

wth possibly another nested fun around that, print_bytes or similar try ing to format it like pryr::object_size. There was an error on "length", something like .data$length would fix ...

**Comment by mdsumner** [comment]
Also see this I think with purrr.

**Comment by mdsumner** [comment]
This is still biting

```R
f2 <- raadfiles:::get_raw_raad_filenames() %>% 
    dplyr::filter(grepl("2013021900_WRF_d2_PCP_f009.nc", file)) %>% 
    dplyr::mutate(fullname = file.path(root, file)) %>% dplyr::pull(fullname)
  tidync(f2) %>% hyper_filter()
  ## not ok
tidync(f2) %>% hyper_filter(x = index < 10)
  
## OK
tidync(f2) %>% hyper_filter(y = index < 10)
 ```

**Comment by mdsumner** [comment]
Fundamental issue is that (x, x = ...) is doomed because the second x clashes with the first. Will call the hyper_filter argument .data. 

Here's some other examples: 

```R
afun <- function(a_input, ...) {
  quo_named <- rlang::quos(...)
  a_input
}

d <- tibble::tibble(a = 1:10, b = head(letters, 10))

afun(d, a_i = index < 2)
Error in afun(d, a_i = index < 2) : object 'index' not found
```

We can trigger the same(?) by jamming the .data argument to mutate: 

```R
data.frame(i = 1) %>% mutate(.da = 1)
Error in UseMethod("mutate_") : 
  no applicable method for 'mutate_' applied to an object of class "c('double', 'numeric')"
```

**Comment by mdsumner** [comment]
There's still more to it, I've put this expectation in a filter test

```R
 expect_error( tidync(f2) %>% hyper_filter(y = index < 10,x = index < 5), "object 'index' not found")

```

**Comment by mdsumner** [comment]
Fixed with dcb94b253b1b45f06f50c05b70deeceb32fe2f2b
